tests/unit/core/test_git_shell.py

```diff
@@ -42,3 +49,138 @@ def test_check_git_dir(tmp_path):
     with cd(str(tmp_path)):
         with pytest.raises(UsageError):
             check_git_dir()
+
+
+def test_check_git_ref(tmp_path):
```

Comment:
This test should be split at least in two: one test for the not-in-a-repo part and another for the in-a-repo part.

There should be only one of GIVEN/WHEN/THEN in each test.